Tests/SonarQube.Bootstrapper.Tests/BuildAgentUpdaterTests.cs

```diff
@@ -68,6 +70,96 @@ public void Updater_CheckDownloadUrl_ConnectionFailure()
             CheckInvalidUrlFails("http://localhost:9000");
         }
 
+        [TestMethod]
+        [Description("The targets file should be copied if none are present. The files should not be copied if they already exist and have not been changed.")]
+        public void InjectTargetsFile_Copy()
+        {
+            // In case the dummy targets file somehow does not get deleted (e.g. when debugging) , make sure its content is valid XML 
+            string sourceTargetsContent = @"<Project ToolsVersion=""4.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"" />";
+            CreateDummySourceTargetsFile(sourceTargetsContent);
+
+            try
+            {
+                InjectTargetsFileAndAssert(sourceTargetsContent, expectCopy: true);
+
+                // if we try to inject again, the targets should not be copied because they have the same content
+                InjectTargetsFileAndAssert(sourceTargetsContent, expectCopy: false);
+            }
+            finally
+            {
+                foreach (string destinationDir in BuildAgentUpdater.DestinationDirs)
+                {
+                    string path = Path.Combine(destinationDir, BuildAgentUpdater.LoaderTargetsName);
+                    File.Delete(path);
+                }
+            }
+        }
+
+        [TestMethod]
+        [Description("The targets should be copied if they don't exist. If they have been changed, the updater should overwrite them")]
+        public void InjectTargetsFile_Overwrite()
+        {
+            // In case the dummy targets file somehow does not get deleted (e.g. when debugging), make sure its content valid XML
+            string sourceTargetsContent1 = @"<Project ToolsVersion=""4.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"" />";
+            string sourceTargetsContent2 = @"<Project ToolsVersion=""12.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"" />";
+
+            CreateDummySourceTargetsFile(sourceTargetsContent1);
+
+            try
+            {
+                InjectTargetsFileAndAssert(sourceTargetsContent1, expectCopy: true);
+                Assert.IsTrue(BuildAgentUpdater.DestinationDirs.Count >= 2, "Expecting at least 2 destination directories");
+
+                string path = Path.Combine(BuildAgentUpdater.DestinationDirs[0], BuildAgentUpdater.LoaderTargetsName);
+                File.Delete(path);
```

Comment:
Seems reasonable but it's not obvious. It would be clearer (and a more rigorous test) if you checked the expected files were overwritten and the deleted file wasn't.